### PR TITLE
fix: Comment instead of autoclose for modified branch

### LINKED ISSUE #3726: Comment - don't autoclose - if a branch has been edited
**What would you like Renovate to be able to do?**

Before autoclosing any branch, Renovate should check if it was the most recent committer. If it wasn't, then this indicates that other commits have probably been added, and Renovate should instead post a comment instead of closing/deleting.

**Describe the solution you'd like**

Compare the last committer of each branch to be autoclosed.

**Describe alternatives you've considered**

Right now Renovate wipes out any branch/PR that it thinks shouldn't be there.

**Additional context**

This will partially mitigate #3220

### PR DESCRIPTION
**Pull request by zharinov**
Closes #3726

### CODE REVIEW

File: lib/workers/repository/finalise/prune.js [hunk 1]
```diff
@@ -26,14 +27,22 @@ async function cleanUpBranches(
...
       const closePr = true;
       logger.info({ branch: branchName }, `Deleting orphan branch`);
-      if (dryRun) {
+      if (!hasSingleAuthor) {
+        logger.info(
+          { prNo: pr.number, prTitle: pr.title },
+          'Skip PR autoclose'
+        );
+        if (pr) {
+          await platform.ensureComment(pr.number, 'FIXME', 'FIXME');
```
Comment (zharinov) [inline lib/workers/repository/finalise/prune.js]: @rarkins

**Comment by zharinov** [comment]
Not sure if I can fix coverage

**Comment by rarkins** [comment]
@zharinov couldn't we just use `getBranchPr()` and then check `pr.canRebase`? It's meant to record whether a PR (branch) has been modified or not. So I think if branchPr exists and branchPr.canRebase is false, then we add a comment, but otherwise (canRebase=true or no PR) then we autoclose.

**Comment by zharinov** [comment]
> @zharinov couldn't we just use `getBranchPr()` and then check `pr.canRebase`? It's meant to record whether a PR (branch) has been modified or not. So I think if branchPr exists and branchPr.canRebase is false, then we add a comment, but otherwise (canRebase=true or no PR) then we autoclose.

Wow, I wasn't thinking this way

**Comment by rarkins** [comment]
@zharinov ready

**Pull request merged**